### PR TITLE
Task list enhancements master

### PR DESCRIPTION
**Pull request by alepintus**
Hi @joewhite101 ,
this is the cherry-pick from [this PR](https://github.com/entando/entando-components/pull/474).

Thanks.

### CODE REVIEW

**Comment by entando-service** [comment]
All is Well

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/apsadmin/portal/specialwidget/BpmTaskListDatatableWidgetAction.java [hunk 1]
```diff
@@ -19,13 +19,32 @@
...
     @Override
-    protected void loadFieldIntoDatatableFromBpm() throws ApsSystemException {
+    protected void loadFieldIntoDatatableFromBpm(String containerId, String procId) throws ApsSystemException {
         HashMap<String, String> opt = new HashMap<>();
         opt.put("user", DEMO_USER);
         KieBpmConfig config = super.getFormManager().getKieServerConfigurations().get(this.getKnowledgeSourcePath());
         List<KieTask> task = super.getFormManager().getHumanTaskList(config, null, opt);
+
+        Set<String> variableFields = super.getFormManager().getProcessVariables(config, containerId, procId);
+        //JPW -- Assign process variables for datatable
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/apsadmin/portal/specialwidget/BpmTaskListDatatableWidgetAction.java]: remove

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/apsadmin/portal/specialwidget/BpmTaskListDatatableWidgetAction.java [hunk 2]
```diff
@@ -19,13 +19,32 @@
...
+//        /server/queries/processes/instances/{pInstanceId}/variables/instances
+//        /server/containers/{id}/processes/definitions/{pId}/variables
+        //JPW -- Assog tasl varoab;es
+
+        List<String> fields = new ArrayList<>();
+        fields.addAll(variableFields);
+
+
+        //Horrible hack. To be replaced by an actual call.
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/apsadmin/portal/specialwidget/BpmTaskListDatatableWidgetAction.java]: Add a TODO tag

**Comment by entando-service** [comment]
All is Well

**Comment by entando-service** [comment]
All is Well

**Pull request merged**